### PR TITLE
Refactor make check

### PR DESCRIPTION
**Pull request by ph**
## What is the problem this PR solves?

This code refactor the `make check` task, it split the task into two different concerns:

-  One is `make check-ci` which is run as part of the Jenkins files, this task deals with code generation, headers check, notice generation and make sure all the changes are checked in the VCS.
- The other is `make check` that run all the checks above and also golangci-lint. The pull request linter is doing using a github action to receive all the inline comments.

// Please do not just reference an issue. Explain WHAT the problem this PR solves here.

## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Requires #1165
- Relates #1161

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-SHELL=/bin/bash
+SHELL=env bash
```
Comment (ph) [inline Makefile]: This is need locally here on my machine, where isolation of packages breaking `/bin/bash`. 
I think this line is not required, but I've used the env` strategy to find where is bash.

File: Makefile [hunk 2]
```diff
@@ -58,25 +58,28 @@ generate: ## - Generate schema models
...
 	@env GOBIN=${GOBIN} go install github.com/elastic/go-licenser@latest
 	@env PATH="${GOBIN}:${PATH}" go-licenser -license Elastic
 
 .PHONY: check-go
-check-go: ## - Run go fmt, go vet, go mod tidy
-	@go fmt ./...
-	@go vet ./...
-	@go mod tidy
+check-go: ## - Run golangci-lint
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.44.2
```
Comment (v1v) [inline Makefile:81]: I wonder if we can use a fixed version instead,for instance the last commit for https://github.com/golangci/golangci-lint/releases/tag/v1.44.2, https://github.com/golangci/golangci-lint/commit/d58dbde584c801091e74a00940e11ff18c6c68bd

```suggestion
	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/d58dbde584c801091e74a00940e11ff18c6c68bd/install.sh | sh -s v1.44.2
```

**Review by v1v — APPROVED** [review]
Only a minor comment regarding the `sha` for the golang-ci-lint tool

File: Makefile (hunk 2 above)
Comment (ph) [reply Makefile:81]: I was using their recommended way in their documentation to integrate with ci tool. I mean we could, but using the tag seems like solid approach.

**Pull request merged**